### ISSUE TITLE
add date picker to forms

### ISSUE THREAD

**Issue by putyourlitterhere** [farmOS/farmOS#324]
Entering dates in forms (e.g. creating log entry, set filter dates, etc.) is cumbersome having at least 3 fields (month, day, year) to fill. It would be easier if there was a date picker input element as for example this one: https://jqueryui.com/datepicker/#inline

**Comment by mstenta** [comment]
Agreed @putyourlitterhere - this is something that I went back and forth on years ago when I made the decision to use the select list dropdowns. At the time, I discovered some bugs with the Drupal date popup widget, and decided that it was better not to depend on it. However, in recent testing it seems to be working well, so I think we should reconsider it.

However, if we change it in one place, we need to change it in ALL places. I'm a stickler for consistency. ;-)

With that in mind, and given the cost/benefit of making that change everywhere in the current codebase (and contrib modules), I have been leaning towards leaving this as-is in farmOS 1.x, and making this change part of the farmOS 2.x roadmap: https://www.drupal.org/project/farm/issues/3092830, specifically in the UI/UX meta roadmap: https://www.drupal.org/project/farm/issues/3151246

I will add a comment there linking back to this issue, as well as another similar issue where we discussed this: https://github.com/farmOS/farmOS/issues/223
